### PR TITLE
[CHORES] setting up library

### PR DESCRIPTION
**Pull request by nofamex**

### CODE REVIEW

File: src/Commands/BadasoLMSSetup.php [hunk 1]
```diff
@@ -35,23 +38,88 @@ public function __construct() {
...
+
+    protected function addingBadasoEnv() {
+        try {
+            $env_path = base_path('.env');
+
+            $env_file = file_get_contents($env_path);
+            $arr_env_file = explode("\n", $env_file);
 
-    protected function linkStorage() {}
+            $env_will_adding = $this->envListUpload();
```
Comment (AldiNFitrah) [inline src/Commands/BadasoLMSSetup.php]: what is this for?

File: src/Commands/BadasoLMSSetup.php [hunk 2]
```diff
@@ -35,23 +38,88 @@ public function __construct() {
...
+    }
+
+    protected function envListUpload() {
+        return [
+            'MIX_LMS_URL_PREFIX' => '/lms',
+            'MIX_FRONTEND_URL' => 'http://localhost:8000',
+        ];
+    }
+
+    protected function addingBadasoEnv() {
```
Comment (AldiNFitrah) [inline src/Commands/BadasoLMSSetup.php]: prefer to use verb1 for function name, e.g. `addBadasoEnv`

File: src/Routes/api.php [hunk 3]
```diff
@@ -1 +1,15 @@
...
+
+$api_route_prefix = config('Badaso.api_route_prefix', 'badaso-api');
+
+Route::group(['prefix' => $api_route_prefix, 'as' => 'badaso.', 'middleware' => [ApiRequest::class]], function() {
+    Route::group(['prefix' => 'module/lms/v1'], function() {
+        Route::group(['prefix' => 'user'], function() {
+            Route::get('/home', HelpersRoute::getController('UserController@home'));
+        });
+    });
+});
```
Comment (AldiNFitrah) [inline src/Routes/api.php]: newline please

File: src/Helpers/Route.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+         // if the key is not found, return $key
+         if (! isset($controllers[$key])) {
+             return 'Uasoft\\Badaso\\Module\\LMS\\Controllers\\'.$key;
+         }
+ 
+         // return the value of the key
+         return $controllers[$key];
+    }
+}
```
Comment (AldiNFitrah) [inline src/Helpers/Route.php]: newline please

File: src/Helpers/Route.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    {
+         // get config 'controllers' from config/badaso-post.php
+         $controllers = config('badaso-lms.controllers');
+
+         // if the key is not found, return $key
+         if (! isset($controllers[$key])) {
+             return 'Uasoft\\Badaso\\Module\\LMS\\Controllers\\'.$key;
+         }
+ 
+         // return the value of the key
```
Comment (AldiNFitrah) [inline src/Helpers/Route.php]: I don't think this comment is useful

File: src/Helpers/Route.php [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+namespace Uasoft\Badaso\Module\LMS\Helpers;
+
+class Route 
+{
+    public static function getController($key)
+    {
+         // get config 'controllers' from config/badaso-post.php
+         $controllers = config('badaso-lms.controllers');
+
+         // if the key is not found, return $key
```
Comment (AldiNFitrah) [inline src/Helpers/Route.php:12]: isn't it pretty obvious?

**Review by Zsaschz — APPROVED** [review]
lms-theme package uses "Uasoft\Badaso\Theme\LMSTheme" as the project namespace. i think we need to synchronize between these two package because they are related

File: src/Commands/BadasoLMSSetup.php (hunk 1 above)
Comment (nofamex) [reply src/Commands/BadasoLMSSetup.php]: updating laravel environment

File: src/Commands/BadasoLMSSetup.php (hunk 2 above)
Comment (nofamex) [reply src/Commands/BadasoLMSSetup.php]: fixed

File: src/Routes/api.php (hunk 3 above)
Comment (nofamex) [reply src/Routes/api.php]: fixed

File: src/Helpers/Route.php (hunk 4 above)
Comment (nofamex) [reply src/Helpers/Route.php]: fixed

File: src/Helpers/Route.php (hunk 5 above)
Comment (nofamex) [reply src/Helpers/Route.php]: removed

File: src/Helpers/Route.php (hunk 6 above)
Comment (nofamex) [reply src/Helpers/Route.php:12]: ok removed

**Review by AldiNFitrah — APPROVED** [review]

**Pull request merged**